src/editors/containers/TextEditor/hooks.test.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import * as module from './hooks';
+
+jest.mock('react', () => {
+  const updateState = jest.fn();
+  return {
+    updateState,
+    useState: jest.fn(val => ([{ state: val }, (newVal) => updateState({ val, newVal })])),
+    createRef: jest.fn(val => ({ ref: val })),
+  };
+});
+
+describe('TextEditor hooks', () => {
+  describe('Editor Init hooks', () => {
+    describe('addImageUploadButton', () => {
+      const mockOpenModal = jest.fn();
+      const mockAddbutton = jest.fn(val => ({ onAction: val }));
+      const editor = {
+        ui: {
+          registry: {
+            addButton: mockAddbutton,
+          },
+        },
+      };
+      let output;
+      beforeEach(() => {
+        output = module.addImageUploadButton(mockOpenModal);
+      });
+      test('It calls addButton in the editor, but openModal is not called', () => {
+        output(editor);
```

Comment:
Should this be done with shallow()?

Reply:
Shallow is for testing rendering behavior-- we're not rendering anything here, as this is testing the behavior of a hook, not a component?

Reply:
I see. That makes sense.